pkg/vpcmodel/output.go

```diff
@@ -301,14 +301,51 @@ func (of *serialOutputFormatter) AggregateVPCsOutput(outputList []*SingleAnalysi
 		}
 		res, err = writeJSON(all, outFile)
 	case Synthesis:
+		connLines := []spec.SpecRequiredConnectionsElem{}
+		externals := spec.SpecExternals{}
+		externalsMap := make(map[string]string)
+		for _, o := range outputList {
+			// always true
+			if structObj, ok := o.jsonStruct.(*spec.Spec); ok {
+				connLines = append(connLines, renameExternals(structObj.RequiredConnections, externalsMap)...)
+				for k, v := range structObj.Externals {
+					externals[externalsMap[k]] = v
+				}
+			}
+		}
 		// in synthesis format we need to follow json spec schema
-		// https://github.com/np-guard/models/blob/main/spec_schema.json
-		// multi-vpc not supported yet
-		res, err = writeJSON(outputList[0].jsonStruct, outFile)
+		res, err = writeJSON(spec.Spec{RequiredConnections: connLines, Externals: externals}, outFile)
 	}
 	return res, err
 }
 
+func renameExternals(requiredConnections []spec.SpecRequiredConnectionsElem,
+	externalsMap map[string]string) []spec.SpecRequiredConnectionsElem {
+	connLines := []spec.SpecRequiredConnectionsElem{}
+	for _, conn := range requiredConnections {
+		if conn.Src.Type == spec.ResourceTypeExternal {
+			if val, ok := externalsMap[conn.Src.Name]; ok {
+				conn.Src.Name = val
+			} else {
+				name := externalString + strconv.Itoa(len(externalsMap))
+				externalsMap[conn.Src.Name] = name
+				conn.Src.Name = name
+			}
+		}
+		if conn.Dst.Type == spec.ResourceTypeExternal {
+			if val, ok := externalsMap[conn.Dst.Name]; ok {
+				conn.Dst.Name = val
+			} else {
+				name := externalString + strconv.Itoa(len(externalsMap))
+				externalsMap[conn.Dst.Name] = name
+				conn.Dst.Name = name
+			}
+		}
```

Comment:
```suggestion
		if conn.Src.Type == spec.ResourceTypeExternal {
			conn.Src.Name = getNewExternalName(conn.Src.Name, externalsMap)
		} else if conn.Dst.Type == spec.ResourceTypeExternal {
			conn.Dst.Name = getNewExternalName(conn.Dst.Name, externalsMap)
		}
```
Avoid code duplication